core/src/main/clojure/xtdb/types.clj

```diff
@@ -704,3 +708,167 @@
 (defn with-nullable-fields [fields]
   (->> fields
        (into {} (map (juxt key (comp #(merge-fields % null-field) val))))))
+
+(defn- read-utf8 [^bytes barr] (String. barr StandardCharsets/UTF_8))
+
+(defn utf8
+  "Returns the utf8 byte-array for the given string"
+  ^bytes [s]
+  (.getBytes (str s) StandardCharsets/UTF_8))
+
+(def pg-types
```

Comment:
worth creating a xtdb.pgwire.types ns for these? 

if nothing else, core doesn't (yet) depends on pgwire, so this creates a (conceptual, not code) cycle

Reply:
although info-schema potentially does already... let's leave these for now, reckon we'll merge pgwire into core soon enough